File: src/main/java/dev/esophose/discordbot/listener/PaginatedEmbedListener.java
package dev.esophose.discordbot.listener;

import dev.esophose.discordbot.Sparky;
import dev.esophose.discordbot.manager.PaginatedEmbedManager;
import discord4j.core.event.domain.message.ReactionAddEvent;

public class PaginatedEmbedListener extends Listener<ReactionAddEvent> {

    private PaginatedEmbedManager paginatedEmbedManager;

    public PaginatedEmbedListener() {
        super(ReactionAddEvent.class);
    }

    @Override
    public void execute(ReactionAddEvent event) {
        if (this.paginatedEmbedManager == null)
            this.paginatedEmbedManager = Sparky.getInstance().getManager(PaginatedEmbedManager.class);

        this.paginatedEmbedManager.changeEmbedState(event);
    }

}

File: src/main/kotlin/dev/esophose/discordbot/listener/PaginatedEmbedListener.kt
package dev.esophose.discordbot.listener

import dev.esophose.discordbot.Sparky
import dev.esophose.discordbot.manager.PaginatedEmbedManager
import discord4j.core.event.domain.message.ReactionAddEvent

class PaginatedEmbedListener : Listener<ReactionAddEvent>(ReactionAddEvent::class.java) {

    private var paginatedEmbedManager: PaginatedEmbedManager? = null

    override fun execute(event: ReactionAddEvent) {
        if (this.paginatedEmbedManager == null)
            this.paginatedEmbedManager = Sparky.getManager(PaginatedEmbedManager::class)

        this.paginatedEmbedManager!!.changeEmbedState(event)
    }

}
